app/controllers/predicates/AuthorisedAction.scala

```diff
@@ -105,23 +119,44 @@ class AuthorisedAction @Inject()()(implicit val authConnector: AuthConnector,
         .withIdentifier(EnrolmentIdentifiers.individualId, identifierId)
         .withDelegatedAuthRule(agentDelegatedAuthRuleKey)
 
-    authorised(agentAuthPredicate(mtdItId))
-      .retrieve(allEnrolments) { enrolments =>
+    val nino: Regex = ".*/nino/(.*)/sources.*".r
 
-        enrolmentGetIdentifierValue(EnrolmentKeys.Agent, EnrolmentIdentifiers.agentReference, enrolments) match {
-          case Some(arn) =>
-            block(User(mtdItId, Some(arn)))
-          case None =>
-            logger.info("[AuthorisedAction][agentAuthentication] Agent with no HMRC-AS-AGENT enrolment.")
-            unauthorized
+    val optionalNino = request.uri match {
+      case nino(group) => Some(group)
```

Comment:
I didn't know you could do this with regex. This is pretty neato.

Reply:
Abdullah taught me the ways